common/src/state/pending_message.rs

```diff
@@ -0,0 +1,182 @@
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use once_cell::sync::Lazy;
+use tokio::sync::{
+    mpsc::{UnboundedReceiver, UnboundedSender},
+    Mutex,
+};
+use uuid::Uuid;
+use warp::{constellation::Progression, crypto::DID};
+
+use crate::warp_runner::ui_adapter::Message;
+
+pub struct AttachmentProgress {
+    pub progress: Progression,
+    pub conversation_id: Uuid,
+    pub msg: PendingSentMessage,
+}
+
+pub struct PendingMessageChannels {
+    pub tx: UnboundedSender<AttachmentProgress>,
+    pub rx: Arc<Mutex<UnboundedReceiver<AttachmentProgress>>>,
+}
+
+pub static MESSAGE_CHANNEL: Lazy<PendingMessageChannels> = Lazy::new(|| {
```

Comment:
see [https://github.com/Satellite-im/Uplink/pull/837/commits/bb3b02a64e31a6497f671da694a42f4c014ad899#diff-e108e69cf0f5933cb78e6fed28b1c742b6ec02401d348c79d689db091b3aefcbR619](https://github.com/Satellite-im/Uplink/pull/837/commits/bb3b02a64e31a6497f671da694a42f4c014ad899#diff-e108e69cf0f5933cb78e6fed28b1c742b6ec02401d348c79d689db091b3aefcbR619) where i merged them with the warp event channel and see if that is fine. we need to handle it separately because else it would constantly update `State` which will hang the app till upload is done

Reply:
resolved in [b403140](https://github.com/Satellite-im/Uplink/pull/837/commits/b40314027f7462cd3318a0705eb0e52caa106092)